src/kontrol/utils.py

```diff
@@ -716,3 +716,144 @@ def decode_log_message(token: str, selector: int) -> str | None:
     238520724: ['address', 'address', 'address', 'bool'],
     1717301556: ['address', 'address', 'address', 'address'],
 }
+
+
+def kontrol_test_file_contents() -> str:
+    return """pragma solidity ^0.8.26;
+import "forge-std/Vm.sol";
+import "forge-std/Test.sol";
+import "kontrol-cheatcodes/KontrolCheats.sol";
+
+contract KontrolTest is Test, KontrolCheats {
+    enum Mode {
+        Assume,
+        Try,
+        Assert
+    }
+    function _establish(Mode mode, bool condition) internal pure returns (bool) {
+        if (mode == Mode.Assume) {
+            vm.assume(condition);
+            return true;
+        } else if (mode == Mode.Try) {
+            return condition;
+        } else {
+            assert(condition);
+            return true;
+        }
+    }
+    function _loadData(
+        address contractAddress,
+        uint256 slot,
+        uint256 offset,
+        uint256 width
+    ) internal view returns (uint256) {
+        require(contractAddress != address(0), 'Invalid contract address');
+        require(width > 0, 'Width must be greater than 0');
+        // `offset` and `width` must not overflow the slot
+        assert(offset + width <= 32);
+        
+        // Slot read mask - handle width = 32 case to prevent overflow
+        uint256 mask;
+        if (width == 32) {
+            mask = type(uint256).max;
+        } else {
+            unchecked {
+                mask = (2 ** (8 * width)) - 1;
+            }
+        }
+        // Value right shift
+        uint256 shift = 8 * offset;
+        // Current slot value
+        uint256 slotValue = uint256(vm.load(contractAddress, bytes32(slot)));
+        // Isolate and return data to retrieve
+        return mask & (slotValue >> shift);
+    }
+    function _storeData(address contractAddress, uint256 slot, uint256 offset, uint256 width, uint256 value) internal {
+        // `offset` and `width` must not overflow the slot
+        assert(offset + width <= 32);
+        // and `value` must fit into the designated part
+        assert(width == 32 || value < 2 ** (8 * width));
+        // Slot update mask
+        uint256 maskLeft;
+        unchecked {
+            maskLeft = ~((2 ** (8 * (offset + width))) - 1);
+        }
```

Comment:
Should we also split this into the `offset + width == 32` case, to make the intention clearer?